test/unit/carousel.ts

```diff
@@ -0,0 +1,414 @@
+          target: {
+            addEventListener: spy(),
+            removeEventListener: spy()
+          }
+        };
+        stub(Carousel, 'isSwipeToNext').returns(true);
+        carousel.moveNext = spy();
+        carousel.movePrevious = spy();
+
+        carousel.onTouchEnd(event as any);
+
+        expect(Carousel.isSwipeToNext).to.be.calledWithExactly({ startX: 0, startY: 5 }, 30, 15);
+        expect(carousel.moveNext).to.be.called;
+        expect(carousel.movePrevious).not.to.be.called;
+        expect(event.target.removeEventListener).to.be.calledWithExactly('touchstart', carousel.onTouchStart);
+        expect(event.target.removeEventListener).to.be.calledWithExactly('touchend', carousel.onTouchEnd);
+      });
+
+      it('should swipe to previous slides if swipe distance is long enough', () => {
+        const event = {
+          changedTouches: [{ pageX: -40, pageY: 15 }],
+          target: {
+            addEventListener: spy(),
+            removeEventListener: spy()
+          }
+        };
+        stub(Carousel, 'isSwipeToNext').returns(false);
+        carousel.moveNext = spy();
+        carousel.movePrevious = spy();
+
+        carousel.onTouchEnd(event as any);
+
+        expect(Carousel.isSwipeToNext).to.be.calledWithExactly({ startX: 0, startY: 5 }, -40, 15);
+        expect(carousel.moveNext).not.to.be.called;
+        expect(carousel.movePrevious).to.be.called;
+        expect(event.target.removeEventListener).to.be.calledWithExactly('touchstart', carousel.onTouchStart);
+      });
+
+      it('should not swipe slides if swipe distance is not long enough', () => {
+        const event = {
+          changedTouches: [{ pageX: 10, pageY: 15 }],
+          target: {
+            addEventListener: spy(),
+            removeEventListener: spy()
+          }
+        };
+        stub(Carousel, 'isSwipeToNext').returns(false);
+        carousel.moveNext = spy();
+        carousel.movePrevious = spy();
+
+        carousel.onTouchEnd(event as any);
+
+        expect(carousel.moveNext).to.not.be.called;
+        expect(carousel.movePrevious).to.not.be.called;
+        expect(event.target.removeEventListener).to.be.calledWithExactly('touchstart', carousel.onTouchStart);
+        expect(event.target.removeEventListener).to.be.calledWithExactly('touchend', carousel.onTouchEnd);
+      });
+    });
+
+    describe('slideHandler()', () => {
+      it('should simply set slide number if speed is 0', () => {
+        const reset = carousel.resetCurrentSlideNum = spy();
+        const update = carousel.set = spy();
+
+        carousel.slideHandler(-1);
+
+        expect(reset).to.be.calledWithExactly(0, -1, 10);
+        expect(update).to.not.be.called;
+      });
+
+      it('should go to the specific slide when threshold is not hit and speed is not 0', () => {
+        carousel.props.speed = 200;
+        const slide = 1;
+        const track = <any>{ addEventListener: spy(), removeEventListener: spy() };
```

Comment:
don't need to cast `track` if you set the type:

*cast to type*:
```js
const track = <any>{ addEventListener: spy(), removeEventListener: spy() };
```

*declare type*:
```js
const track: any = { addEventListener: spy(), removeEventListener: spy() };
```